src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -308,19 +308,42 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                 return null;
             }
 
-            SNIHandle sniHandle = null;
+            if (isIntegratedSecurity)
+            {
+                string hostName = details.ServerName;
+                int connPort = details.Port;
+                string connInstanceName = details.InstanceName;
+
+                try
+                {
+                    if (connPort >= 0)
+                    {
+                        spnBuffer = GetSqlServerSPN(hostName, connPort);
+                    }
+                    else
+                    {
+                        spnBuffer = GetSqlServerSPN(hostName, connInstanceName);
+                    }
+                }
+                catch (Exception e)
+                {
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.TCP_PROV, SNICommon.InvalidConnStringError, e);
```

Comment:
`SNIError` constructor is 
```public SNIError(SNIProviders provider, uint sniErrorCode, Exception sniException)```
I will change it to `new SNIError(SNIProviders.INVALID_PROV, SNICommon.InvalidConnStringError, e);`